### ISSUE TITLE
org.apache.avro.SchemaParseException: Can't redefine: <schema-name>

### ISSUE THREAD

**Issue by mnarrell** [julianpeeters/avrohugger#73]
I have two or more schema files containing common objects.  When I try to generate Specific classes for them, i get:
```
[info] Compiling AVSC /Users/matt/schema1.avsc
[info] Compiling AVSC /Users/matt/schema2.avsc
org.apache.avro.SchemaParseException: Can't redefine: <a common record between the two>
        at org.apache.avro.Schema$Names.put(Schema.java:1060)
        at org.apache.avro.Schema$Names.add(Schema.java:1055)
        at org.apache.avro.Schema.parse(Schema.java:1187)
        at org.apache.avro.Schema.parse(Schema.java:1272)
        at org.apache.avro.Schema.parse(Schema.java:1203)
        at org.apache.avro.Schema$Parser.parse(Schema.java:965)
        at org.apache.avro.Schema$Parser.parse(Schema.java:932)
        at avrohugger.input.parsers.FileInputParser.getSchemaOrProtocols(FileInputParser.scala:54)
        at avrohugger.FileGenerator$.fileToFile(FileGenerator.scala:82)
        at avrohugger.Generator.fileToFile(Generator.scala:81)
        at sbtavrohugger.FileWriter$$anonfun$generateCaseClasses$1.apply(FileWriter.scala:21)
        at sbtavrohugger.FileWriter$$anonfun$generateCaseClasses$1.apply(FileWriter.scala:19)
        at scala.collection.LinearSeqOptimized$class.foreach(LinearSeqOptimized.scala:60)
        at scala.collection.mutable.MutableList.foreach(MutableList.scala:30)
        at sbtavrohugger.FileWriter$.generateCaseClasses(FileWriter.scala:19)
        at sbtavrohugger.formats.specific.SpecificGeneratorTask$$anonfun$specificCaseClassGeneratorTask$1$$anonfun$1.apply(SpecificGeneratorTask.scala:43)
        at sbtavrohugger.formats.specific.SpecificGeneratorTask$$anonfun$specificCaseClassGeneratorTask$1$$anonfun$1.apply(SpecificGeneratorTask.scala:40)
        at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:253)
        at sbt.FileFunction$$anonfun$cached$1.apply(Tracked.scala:253)
        at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:267)
        at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3$$anonfun$apply$4.apply(Tracked.scala:263)
        at sbt.Difference.apply(Tracked.scala:224)
        at sbt.Difference.apply(Tracked.scala:206)
        at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:263)
        at sbt.FileFunction$$anonfun$cached$2$$anonfun$apply$3.apply(Tracked.scala:262)
        at sbt.Difference.apply(Tracked.scala:224)
        at sbt.Difference.apply(Tracked.scala:200)
        at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:262)
        at sbt.FileFunction$$anonfun$cached$2.apply(Tracked.scala:260)
        at sbtavrohugger.formats.specific.SpecificGeneratorTask$$anonfun$specificCaseClassGeneratorTask$1.apply(SpecificGeneratorTask.scala:45)
        at sbtavrohugger.formats.specific.SpecificGeneratorTask$$anonfun$specificCaseClassGeneratorTask$1.apply(SpecificGeneratorTask.scala:37)
        at scala.Function8$$anonfun$tupled$1.apply(Function8.scala:35)
        at scala.Function8$$anonfun$tupled$1.apply(Function8.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
```
Is there anyway to work around this?

**Comment by julianpeeters** [comment]
Hi Matt, yes, you can pull out the common record definition, put it in it's own file, and reference it by it's fully-qualified name (e.g. https://github.com/julianpeeters/avrohugger/blob/master/avrohugger-filesorter/src/test/resources/avro/a.avsc)

**Comment by mnarrell** [comment]
Well, yes.  However the schema registries usually deliver an un-factored schema.  I suppose this is a limitation of Avro itself and less of this project.  I appreciate your work here.

**Comment by julianpeeters** [comment]
@mnarrell apologies I didn't catch this earlier, but in case this happens to still be of interest to you, I believe  this issue is solved by smarter usage of the schema parser, available in avrohugger version 1.0.0-RC4 and sbt-avrohugger 2.0.0-RC4
